apps/condo/domains/news/components/RecipientCounter.tsx

```diff
@@ -0,0 +1,195 @@
+    if (newsItemScopes.length === 0) {
+        return intl.formatMessage({ id: 'news.component.RecipientCounter.toResidentsInAllProperties' })
+    } else if (newsItemScopes.length === 1 && isTargetedToEntireProperty(newsItemScopes[0])) {
+        return intl.formatMessage({ id: 'news.component.RecipientCounter.toResidentsInProperty' }, {
+            address: newsItemScopes[0].property.address,
+        })
+    } else if (every(newsItemScopes, isTargetedToEntireProperty)) {
+        const displayCount = 3
+        const addressList = newsItemScopes.map(({ property }) => property.name).join(', ')
+        const andMoreCount = newsItemScopes.length <= displayCount ? null : newsItemScopes.length - displayCount
+        const andMore = !andMoreCount ? '' : intl.formatMessage({ id: 'news.component.RecipientCounter.toResidentsInProperties.andMore' }, { count: andMoreCount })
+        return intl.formatMessage({ id: 'news.component.RecipientCounter.toResidentsInProperties' }, {
+            addressList,
+            andMore,
+        })
+    } else if (areTargetedToOneProperty(newsItemScopes) && every(newsItemScopes, isTargetedToUnitName)) {
+        const property: PropertyType = newsItemScopes[0].property
+        const targetedSections = detectTargetedSections(newsItemScopes, property)
+        if (targetedSections.length === 1) {
+            return intl.formatMessage({ id: 'news.component.RecipientCounter.toResidentsInPropertySection' }, {
+                section: targetedSections[0].name,
+                address: newsItemScopes[0].property.address,
+            })
+        } else if (targetedSections.length > 1) {
+            return intl.formatMessage({ id: 'news.component.RecipientCounter.toResidentsInPropertySections' }, {
+                sections: map(targetedSections, 'name').join(', '),
+                address: newsItemScopes[0].property.address,
+            })
+        } else {
+            return intl.formatMessage({ id: 'news.component.RecipientCounter.toResidentsInPropertyUnits' }, {
+                unitNames: map(newsItemScopes, 'unitName').join(', '),
+                address: newsItemScopes[0].property.address,
+            })
+        }
+    }
+}
+
+/**
+ * Will subtract units, covered by "residents" from units covered by "properties"
+ * @param residents - Residents, that will receive NewsItem, scoped by NewsItemScope records
+ * @param properties - Properties, covered by NewsItemScope records
+ */
+const calculateWillNotReceiveCount = (residents: ResidentType[], properties: PropertyType[]) => {
+    return properties.reduce((acc, property) => {
+        const propertyResidents = filter(residents, { property: { id: property.id } })
+        const propertyUnits = getUnitsFromProperty(property)
+        const residentUnits = uniq(map(propertyResidents, 'unitName'))
+        return acc + difference(propertyUnits, residentUnits).length
+    }, 0)
+}
+
+interface RecipientCounterProps {
+    newsItemScopes: NewsItemScope[]
```

Comment:
may be `NewsItemScopeWhereInput`. At the time of the news creation, there is no scope instance yet and it is impossible to determine the dv sender id which is mandatory in this type

Reply:
But it is better for the front to describe a special interface with the expected fields

Reply:
The new type was added